### PR TITLE
Color page first draft

### PR DESCRIPTION
**Pull request by NadiemM**
#### Color page basics should be finished
- code probably not optimal yet (feedback would be awesome)

#### Add-ons:
- Added an `intersectionObserver` named it `animationObserver`, to add some animations on scroll to the page
- Since we're removing the scrollbar from the overflowing grid, moving the grid without a trackpad is not possible (similar on the typography page). Added `draggable.ts` Now it should be possible to scroll left and right by dragging the grid. 
- Selecting a color and setting it as an app color will automatically change the CSS variables

#### Future to-do's:

Main priority:
- Optimize code and structure
- Add Correct dropdown/sidedown menu component to use within the page

Decent priority:
- Create the badge as component and use it on the color page
- Find a solution to store the changes to the CSS variables, something like storing them in local storage and pulling them out. To not lose the CSS variables set on page reload

Far away priorities:
- Add a reset button, resets all app colors to initial colors
- Get config button: pop-up with a tailwind.config.js file structure and content, will contain all the colors the user has set to overwrite our styles
- Create the colorCard component and use it on the color page
- Specify animations and talk about which ones will be definitive or what the new ones can look like
- Optimize drag function (if needed)

### CODE REVIEW

File: static/tailwind-icon.svg
Comment (benjaminpreiss) [inline static/tailwind-icon.svg:1]: This icon should go into the $lib/assets/icons folder, named exactly as the asset is named in tailwind and can then be imported using `import code from '$lib/assets/icons/Code.svg?url'` and used as `<div style={'background-image: url('${code}')'}/>`

**Review by benjaminpreiss — COMMENTED** [review]
@nadiem some additional feedback

File: src/app.css [hunk 1]
```diff
@@ -12,3 +12,8 @@
 		display: none; /* for Chrome, Safari, and Opera */
 	}
 }
+
+html {
```
[the remark this answers is not in the document]
Comment (NadiemM) [reply src/app.css:16]: Standard classes I always add to a page, to add to the html class, smooth scroll when scrolling to an anchor and not displaying a weird blue shade when clicking a button on mobile

File: src/lib/tailwindPlugin.ts [hunk 2]
```diff
@@ -544,6 +516,7 @@ export default plugin(
 		theme: {
 			extend: {
 				colors: {
+					// this still gives a type error because of the opacity within the getVarColorFunction
 					untld
```
Comment (NadiemM) [inline src/lib/tailwindPlugin.ts:520]: Can't seem to fix this type error

**Comment by NadiemM** [comment]
- [ ]  exchange shadow

**Pull request merged**